### PR TITLE
fix(aws): link integration to v alias of function

### LINKED ISSUE #399: Make integration for `v{N}` go to function alias `v{N}`
Right now, when a new function symbolic `X.Y.Z` is created, an API Gateway integration is installed that routes the external address `/helix-services/function/vX` to function `helix-services--function:X_Y_Z`. Instead, it could route to function alias `helix-services--function:vX`, which would make a lot of integration installations obsolete.

/cc: @tripodsan

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #399

### CODE REVIEW

File: src/deploy/AWSDeployer.js [hunk 1]
```diff
@@ -627,21 +627,23 @@ export default class AWSDeployer extends BaseDeployer {
...
       }));
       this.log.info(chalk`--: updating alias {blue ${name}}...`);
-      await this._lambda.send(new UpdateAliasCommand({
+      const res = await this._lambda.send(new UpdateAliasCommand({
         FunctionName: functionName,
         Name: name,
         FunctionVersion: functionVersion,
       }));
       this.log.info(chalk`{green ok:} updated alias {blue ${name}} to version {yellow ${functionVersion}}.`);
+      return res.AliasArn;
```
Comment (dominique-pfister) [inline src/deploy/AWSDeployer.js:636]: Return the alias ARN updated or created

File: src/deploy/AWSDeployer.js [hunk 2]
```diff
@@ -654,81 +656,73 @@ export default class AWSDeployer extends BaseDeployer {
...
+      // find or create integration
+      let integration = await this.findIntegration(ApiId, aliasArn);
+      if (integration) {
+        this.log.info(`--: using existing integration "${integration.IntegrationId}" for "${aliasArn}"`);
+      } else {
+        integration = await this._api.send(new CreateIntegrationCommand({
+          ApiId,
+          IntegrationMethod: 'POST',
+          IntegrationType: 'AWS_PROXY',
+          IntegrationUri: aliasArn,
```
Comment (dominique-pfister) [inline src/deploy/AWSDeployer.js:694]: point integration to v(M.m) or V(M) alias for function

File: test/aws.integration.js [hunk 3]
```diff
@@ -91,23 +90,26 @@ describe('AWS Integration Test', () => {
 
     const res = await builder.run();
     assert.ok(res);
-    const out = builder.cfg._logger.output;
-    assert.ok(/.*deleted \d+ unused integrations.*/sg.test(out), out);
```
Comment (dominique-pfister) [inline test/aws.integration.js:95]: cleanup is no longer implicit

File: test/aws.integration.js [hunk 4]
```diff
@@ -163,7 +165,21 @@ describe('AWS Integration Test', () => {
...
     console.log('invoking w/o token should fail');
-    let ret = await fetch('https://lqmig3v5eb.execute-api.us-east-1.amazonaws.com/simple-package/simple-name/ci/foo');
+    let ret;
+
+    for (let tries = 3; tries >= 0; tries -= 1) {
+      // eslint-disable-next-line no-await-in-loop
+      ret = await fetch('https://lqmig3v5eb.execute-api.us-east-1.amazonaws.com/simple-package/simple-name/ci/foo');
+      if (ret.status !== 401) {
+        // eslint-disable-next-line no-console
+        console.log(`!!: ${ret.status} !== 401 (retry)`);
```
Comment (dominique-pfister) [inline test/aws.integration.js:175]: reusing an route+integration might lead to requests initially bypassing the authorizer

**Review by tripodsan — APPROVED** [review]

**Pull request merged**